src/middlewares/types.ts

```diff
@@ -0,0 +1,11 @@
+import { AnyAction } from 'combine-reducers'
+import { GlobalState } from 'reducers-actions'
+
+/**
+ * Middleware
+ * next: React.Dispatch<AnyAction> === the NEXT dispatch in the middleware thru-put
+ */
+export type Middleware = (
+  getState: () => GlobalState,
+  dispatch: React.Dispatch<AnyAction>,
+) => (next: React.Dispatch<AnyAction>) => (action: AnyAction) => void
```

Comment:
This sentence is causing me brainfreeze

Reply:
Yeah it's the curried AF redux way, basically setup inside useGlobalState uses the first params getState and dispatch to pass state and dispatch, then next is the "next" Middleware in the pipeline, with action being obv